src/Surfnet/ServiceProviderDashboard/Domain/Entity/Entity.php

```diff
@@ -48,6 +48,9 @@ class Entity
     const STATE_DRAFT = 'draft';
     const STATE_PUBLISHED = 'published';
 
+    const TYPE_SAML = 'saml20';
```

Comment:
Shouldn't it be better if a ValueObject is used for re-usability on the long term? For example an EntityTypeObject which could be used throughout the project? Typehinting support would be much better and you could prevent adding logic in concrete implementations using the type of an entity.

Reply:
Yes, that would be much better, and during the design stage of the entities, we also considered this. The biggest reason we decided against this is the way Doctrine works with custom types.